prov/tcp/src/tcpx_attr.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2017 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "tcpx.h"
+
+
+struct fi_tx_attr tcpx_tx_attr = {
+	.caps = FI_MSG | FI_SEND,
+	.comp_order = FI_ORDER_STRICT,
+	.inject_size = 64,
+	.size = TCPX_MAX_MSG_SIZE,
+	.iov_limit = TCPX_IOV_LIMIT
+};
+
+struct fi_rx_attr tcpx_rx_attr = {
+	.caps = FI_MSG | FI_RECV,
+	.comp_order = FI_ORDER_STRICT,
+	.total_buffered_recv = 0,
+	.size = TCPX_MAX_MSG_SIZE,
+	.iov_limit = TCPX_IOV_LIMIT
+};
+
+struct fi_ep_attr tcpx_ep_attr = {
+	.type = FI_EP_MSG,
+	.protocol = FI_PROTO_SOCK_TCP,
+	.protocol_version = 0,
+	.max_msg_size = TCPX_MAX_MSG_SIZE,
+	.tx_ctx_cnt = 1,
+	.rx_ctx_cnt = 1
+};
+
+struct fi_domain_attr tcpx_domain_attr = {
+	.name = "tcp",
+	.threading = FI_THREAD_SAFE,
+	.control_progress = FI_PROGRESS_AUTO,
+	.data_progress = FI_PROGRESS_AUTO,
+	.resource_mgmt = FI_RM_ENABLED,
+	.av_type = FI_AV_UNSPEC,
+	.mr_mode = FI_MR_UNSPEC,
+	.cq_cnt = 256,
+	.ep_cnt = 1024,
+	.tx_ctx_cnt = 256,
+	.rx_ctx_cnt = 256,
```

Comment:
tc_ctx_cnt and rx_ctx_cnt should match the ep_cnt.  For scaling, we'll want these values higher.  The ep_cnt will limit the number of peers that we can connect to.  This will ultimately be limited by the number of fd's that the app can open.  I'm not sure if you can dynamically get the value, but maybe bump this to 4k or 8k.  That will be higher than the fd limit by default.